templating/twig_extension.rst

```diff
@@ -58,14 +58,114 @@ Create a class that extends ``\Twig_Extension`` and fill in the logic::
     `global variables`_.
 
 Register an Extension as a Service
-----------------------------------
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Next, register your class as a service and tag it with ``twig.extension``. If you're
 using the :ref:`default services.yml configuration <service-container-services-load-example>`,
 you're done! Symfony will automatically know about your new service and add the tag.
 
 You can now start using your filter in any Twig template.
 
+Creating Lazy-Loaded Twig Extensions
+------------------------------------
+
+.. versionadded:: 1.26
+    Support for lazy-loaded extensions was introduced in Twig 1.26.
+
+Including the code of the custom filters/functions in the Twig extension class
+is the simplest way to create extensions. However, Twig must initialize all
+extensions before rendering any template, even if the template doesn't use an
+extension. This means that performance can be slow if you define extensions with
+lots of complex dependencies (e.g. those making database connections).
+
+That's why Twig allows to decouple the extension definition from its
+implementation. Following the same example as before, the first change would be
+to remove the ``priceFilter()`` method from the extension and update the PHP
+callable defined in ``getFilters()``::
+
+    // src/AppBundle/Twig/AppExtension.php
+    namespace AppBundle\Twig;
+
+    use AppBundle\Twig\AppRuntime;
+
+    class AppExtension extends \Twig_Extension
+    {
+        public function getFilters()
+        {
+            return array(
+                // the logic of this filter is now implemented in a different class
+                new \Twig_SimpleFilter('price', array(AppRuntime::class, 'priceFilter')),
+            );
+        }
+    }
+
+Then, create the new ``AppRuntime`` class (it's not required but these classes
+are suffixed with ``Runtime`` by convention) and include the logic of the
+previous ``priceFilter()`` method::
+
+    // src/AppBundle/Twig/AppRuntime.php
+    namespace AppBundle\Twig;
+
+    class AppRuntime
+    {
+        // for complex Twig extensions you may need to define a constructor
```

Comment:
I would even say that if you don't have dependencies, you should never create a runtime.

Reply:
I agree. I've reworded some things. Thanks!